tests/Test/Xrefcheck/TooManyRequestsSpec.hs

```diff
@@ -0,0 +1,59 @@
+{- SPDX-FileCopyrightText: 2021 Serokell <https://serokell.io>
+ -
+ - SPDX-License-Identifier: MPL-2.0
+ -}
+
+module Test.Xrefcheck.TooManyRequestsSpec where
+
+import Universum
+
+import Control.Concurrent (forkIO, killThread)
+import qualified Control.Exception as E
+import qualified Data.CaseInsensitive as CI
+import qualified Data.Map as M
+import Network.HTTP.Types (Status (..), ok200, serviceUnavailable503, tooManyRequests429)
+import Network.HTTP.Types.Header (hRetryAfter)
+import Test.Hspec (Spec, describe, it, shouldBe)
+import Web.Firefly (ToResponse (toResponse), route, run)
+
+import Xrefcheck.Config
+import Xrefcheck.Core
+import Xrefcheck.Verify
+
+spec :: Spec
+spec = do
+  describe "429 response tests" $ do
+    it "Returns 200 eventually" $
+      checkLinkWithServer (mock429 True)
+        "http://127.0.0.1:5000/429" $ VerifyResult []
+    it "Returns 503 eventually" $
+      checkLinkWithServer (mock429 False)
+        "http://127.0.0.1:5000/429" $ VerifyResult
+          [ ExternalHttpResourceUnavailable $
+              Status { statusCode = 503, statusMessage = "Service Unavailable"}
+          ]
+  where
+    checkLinkWithServer mockServer link expectation =
+      E.bracket (forkIO (newIORef 0 >>= mockServer)) killThread $ \_ -> do
+        verifyReferenceDef link >>= (`shouldBe` expectation)
+
+    verifyReferenceDef link = do
+      progRef <- newIORef $ initVerifyProgress []
+      fmap wrlItem <$> verifyReference
+        ((cVerification $ defConfig GitHub) { vcCheckLocalhost = True }) FullMode
+        progRef (RepoInfo M.empty) link ""
+        (Reference "" (toText link) Nothing (Position Nothing))
+
+    mock429 :: Bool -> IORef Int -> IO ()
+    mock429 mode switchRef = run 5000 $ do
```

Comment:
It's not immediately clear what `mode` means here. Maybe passing a `Status` instead of a `Bool` as a parameter would make it clearer.